homeassistant/components/concord232/alarm_control_panel.py

```diff
@@ -124,14 +134,41 @@ def alarm_arm_home(self, code: str | None = None) -> None:
             self._alarm.arm("stay", "silent")
         else:
             self._alarm.arm("stay")
+        self.wait_for_state(STATE_ALARM_ARMED_HOME)
 
     def alarm_arm_away(self, code: str | None = None) -> None:
         """Send arm away command."""
         if not self._validate_code(code, STATE_ALARM_ARMED_AWAY):
             return
         self._alarm.arm("away")
+        self.wait_for_state(STATE_ALARM_ARMED_AWAY)
+
+    def wait_for_state(self, target_state: str, timeout: int = 5) -> None:
+        """Wait for alarm panel to arm/disarm."""
+        if target_state == STATE_ALARM_DISARMED:
+            self._attr_state = STATE_ALARM_DISARMING
+        else:
+            self._attr_state = STATE_ALARM_ARMING
+
+        self.async_write_ha_state()
+
+        timeout_time = time.time() + timeout
+
+        while True:
+            time.sleep(0.5)
+            if self.alarm_status == target_state:
+                break
+            if time.time() > timeout_time:
+                _LOGGER.error(
+                    "Timed out waiting for '%(t)s'. Current state is '%(c)s'",
+                    {"t": target_state, "c": self._attr_state},
+                )
+
+                break
+
+        self._attr_state = self.alarm_status
 
-    def _validate_code(self, code, state):
+    def _validate_code(self, code, state) -> bool:
```

Comment:
Partially typed method signature. Either type the whole signature or don't type it at all.